src/main.cpp

```diff
@@ -4309,6 +4315,13 @@ bool IsCommunityFundEnabled(const CBlockIndex* pindexPrev, const Consensus::Para
     return (VersionBitsState(pindexPrev, params, Consensus::DEPLOYMENT_COMMUNITYFUND, versionbitscache) == THRESHOLD_ACTIVE);
 }
 
+bool IsCommunityFundAccumulationEnabled(const CBlockIndex* pindexPrev, const Consensus::Params& params, bool fStrict)
+{
+    LOCK(cs_main);
+    return (IsCommunityFundEnabled(pindexPrev, params) && !fStrict) ||
+          (VersionBitsState(pindexPrev, params, Consensus::DEPLOYMENT_COMMUNITYFUND_ACCUMULATION, versionbitscache) == THRESHOLD_ACTIVE);
```

Comment:
Also why do we have fStrict? it seems to be an exit for when we are connecting block. but wouldn't it be better to check it each time? - or is it a performance issue?

Just tying to understand?

Reply:
when fStrict is true, IsCommunityFundAccumulationEnabled checks only for DEPLOYMENT_COMMUNITYFUND_ACCUMULATION soft fork.
when false checks for this _OR_ DEPLOYMENT_COMMUNITYFUND